client/dashboard/task-list/tasks/payments/stripe.js

```diff
@@ -8,124 +8,149 @@ import apiFetch from '@wordpress/api-fetch';
+import { getAdminLink } from '@woocommerce/wc-admin-settings';
+import { getQuery } from '@woocommerce/navigation';
 import { WCS_NAMESPACE } from 'wc-api/constants';
-import { recordEvent } from 'lib/tracks';
+import withSelect from 'wc-api/with-select';
+
+/**
+ * Internal dependencies
+ */
+import { getCountryCode } from 'dashboard/utils';
 
 class Stripe extends Component {
 	constructor( props ) {
 		super( props );
 
 		this.state = {
-			errorMessage: '',
-			connectURL: '',
-			showConnectionButtons:
-				! props.manualConfig && ! props.createAccount,
-			showManualConfiguration: props.manualConfig,
+			autoConnectFailed: false,
+			connectURL: null,
+			errorTitle: null,
+			errorMessage: null,
+			isPending: true,
 		};
 
+		this.autoCreateAccount = this.autoCreateAccount.bind( this );
 		this.updateSettings = this.updateSettings.bind( this );
 	}
 
 	componentDidMount() {
-		const { createAccount, options } = this.props;
-		const { showConnectionButtons } = this.state;
-
+		const { stripeSettings } = this.props;
 		const query = getQuery();
 
 		// Handle redirect back from Stripe.
 		if ( query[ 'stripe-connect' ] && query[ 'stripe-connect' ] === '1' ) {
-			const stripeSettings = get(
-				options,
-				[ 'woocommerce_stripe_settings' ],
-				[]
-			);
 			const isStripeConnected =
 				stripeSettings.publishable_key && stripeSettings.secret_key;
 
 			if ( isStripeConnected ) {
-				recordEvent( 'tasklist_payment_connect_method', {
-					payment_method: 'stripe',
-				} );
-				this.props.markConfigured( 'stripe' );
-				this.props.createNotice(
-					'success',
-					__( 'Stripe connected successfully.', 'woocommerce-admin' )
-				);
+				this.completeMethod();
 				return;
 			}
 
 			/* eslint-disable react/no-did-mount-set-state */
 			this.setState( {
-				showConnectionButtons: false,
-				showManualConfiguration: true,
+				autoConnectFailed: true,
 			} );
 			/* eslint-enable react/no-did-mount-set-state */
-			return;
 		}
 
-		if ( createAccount ) {
-			this.autoCreateAccount();
-		}
-
-		if ( showConnectionButtons ) {
+		if ( ! this.requiresManualConfig() ) {
 			this.fetchOAuthConnectURL();
 		}
 	}
 
-	componentDidUpdate( prevProps, prevState ) {
-		if (
-			prevState.showConnectionButtons === false &&
-			this.state.showConnectionButtons
-		) {
-			this.fetchOAuthConnectURL();
+	componentDidUpdate( prevProps ) {
+		const {
+			createNotice,
+			isOptionsRequesting,
+			hasOptionsError,
+		} = this.props;
+
+		if ( prevProps.isOptionsRequesting && ! isOptionsRequesting ) {
+			if ( ! hasOptionsError ) {
+				this.completeMethod();
+			} else {
+				createNotice(
+					'error',
+					__(
+						'There was a problem saving your payment setings',
+						'woocommerce-admin'
+					)
+				);
+			}
 		}
 	}
 
+	requiresManualConfig() {
+		const { activePlugins, isJetpackConnected } = this.props;
+		const { autoConnectFailed } = this.state;
+
+		return isJetpackConnected &&
```

Comment:
Minor but this ternary is kind of hard to follow being on multiple lines.

Reply:
Updated in https://github.com/woocommerce/woocommerce-admin/pull/3782/commits/a7dc0d16d2b7a9a2bdc505b7b1a24bbb81ad0553 - let me know if that looks a bit easier to read.